### ISSUE TITLE
Command line logs lag behind

### ISSUE THREAD

**Issue by HeadLessHUN** [CloudCompare/CloudCompare#1917]
Hello i faced this issue since a long time that the log file or the command line logs or lagging behind, and probably on a lower priority thread, or something like that. And updated intermittently.

Expected behaviour:
It is expected to be forced update on every argument processed.

Env:
Windows 10 x64
Latest alpha and stable version

**Comment by dgirardeau** [comment]
Yes t's the default behavior of the console/log (to process the messages in a separate thread with lower priority).

One would have to add an option to control this behavior...

**Comment by HeadLessHUN** [comment]
Is it takes so much resource worth making it low priority? 
I'm little bit lost where it calls the thread with low priority i see a call stack list like this:

main.cpp
->ccCommandLineParser.cpp
-->ccConsole.cpp

auto refresh is set to true and uses a 1000 (i suppose it is defined in miliseconds), but the console clearly not updates in every second, there are times  when it does not update for hours (loading multiple bigger files).

**Comment by dgirardeau** [comment]
Ah ok, so it's actually the main thread, and the low priority is just this timer that is triggered (at best) every 1000ms.

But since almost everything is done on the main thread in the command line mode (or at least each command is 'blocking'), the console can't refresh itself...

**Comment by HeadLessHUN** [comment]
This is the last thing, which annoys me :D

![image](https://github.com/CloudCompare/CloudCompare/assets/10095691/b620aa7b-3089-40bf-a1da-0b0f76bf1d84)

but i didn't find any useful information... maybe if we would let the app to process events a little bit more between commands ([processEvents](https://doc.qt.io/qt-6/qcoreapplication.html#processEvents-1)(QEventLoop::ProcessEventsFlags flags, int ms)). but of course it would be better to run the commands as soon as it was deployed so probably on the main thread. not on a secondary. I know it's probably not gonna work for the gui.

**Comment by dgirardeau** [comment]
So the issue is more that everything happens on the main thread (both the processing/execution of the commands and the remaining GUI) and it can prevent the console from refreshing if it takes too long.

I've added an option to bypass the timer-based refresh: https://github.com/CloudCompare/CloudCompare/pull/1936

This would not work with the GUI version (as it can cause some OpenGL issues) but I believe it should be ok in the CLI mode. Can you test it?

**Comment by HeadLessHUN** [comment]
well i cannot get you back with good results. here is a test:

-O
-O
-O
...
-MERGE_CLOUDS
-O
-O
-O
...
-MERGE_CLOUDS

And maybe it looks worse then before :/ it only refreshes once all of them is finished. 

I've used xyz (ascii) files to test, because that is slow enough load. bin files loads almost instantly with the same small files

**Comment by dgirardeau** [comment]
Hum, the only thing that is missing in 'silent' mode is all the calls to the GUI (and probably the calls to 'processEvents' by the progress dialog). I've replaced the call to 'repaint' by processEvents'. Can you retry?

**Comment by HeadLessHUN** [comment]
same -SILENT hangs till the end of the process
gui console looks fine.

But in the gui console mode there are lot of processEvents() calls sometimes even every line of point reading... in the IOFilters.

**Comment by HeadLessHUN** [comment]
https://github.com/CloudCompare/CloudCompare/blob/a868dff9ea0c938639df796eaf22f725254d253c/qCC/ccConsole.cpp#L343

can you change the above line to this if you are already working on the ccConsole?
`		printf("%s\n", qPrintable(formatedMessage));`

**Comment by dgirardeau** [comment]
So I'm not sure to understand. I don't use the system console (on Windows, it doesn't appear). Were you referring to the system console/terminale?

**Comment by HeadLessHUN** [comment]
I'm using gitbash usually, but i suppose the same happens with windows system console, but yes, the printf messages and the log files only updates not regurarily, or the end of the process. So -SILENT mode still not updating as it should.

**Comment by dgirardeau** [comment]
Ok, because what I did was only for the GUI console 😅 

I guess for the 'printf' calls, it's just a matter of flushing the system console?

**Comment by HeadLessHUN** [comment]
I'm sorry but i don't care about that :P

I don't really know how these works. Its a little bit new for me, that is why i was not able to do this :)

**Comment by dgirardeau** [comment]
Ok I guess I can get rid of that PR 😢. Your initial message was quite misleading as you were not speaking of the silent mode...

I added a line to remove the buffering of stdout (printing to the system console). I'll let you try, and if it works, I'll replace this PR by something much more simple. And if not...

**Comment by HeadLessHUN** [comment]
Yep it works as it should right now.

I think we can keep the gui console updates as well.I think it works better there as well :) 

Yes sorry I never use gui console. i forgot about that :) But is it worth keeping that gui console version at all?

**Comment by HeadLessHUN** [comment]
> https://github.com/CloudCompare/CloudCompare/blob/a868dff9ea0c938639df796eaf22f725254d253c/qCC/ccConsole.cpp#L343
> 
> can you change the above line to this if you are already working on the ccConsole? ` printf("%s\n", qPrintable(formatedMessage));`

can you add this as well to this pr?

**Comment by dgirardeau** [comment]
For the console, I'm not sure it was that necessary (the timing could simply be reduced from 1 sec to 0.2 secs for instance). These changes are quite big and they are conflicting with OpenGL (so it's a bit 'dangerous' in case this is misused in the future).

And yes, the console is necessary because depending on how you run your command, you might have a system console/terminal visible.

**Comment by HeadLessHUN** [comment]
If you feel it safer to remove these changes then ok. With the setbuf changes it works quite good for me :)

**Comment by dgirardeau** [comment]
Ok I've updated the PR, if you can test that it still works as expected that would be useful.

The silent mode should work as with the last commit, and the GUI console mode just refreshes every 200ms instead of 1sec.

**Comment by HeadLessHUN** [comment]
system console looks ok, gui console also.
